src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,489 @@
+
+  /**
+   * The set of classes that are specified by the user for Randoop to consider. These are classes
+   * supplied in the command line arguments (e.g. --classlist).
+   */
+  private static Set<@ClassGetName String> SPECIFIED_CLASSES =
+      GenInputsAbstract.getClassnamesFromArgs(AccessibilityPredicate.IS_ANY);
+
+  /**
+   * The set of classes that demand-driven uses to generate inputs but are not specified by the
+   * user.
+   */
+  private static Set<Class<?>> unspecifiedClasses = new LinkedHashSet<>();
+
+  /** True if an exact type match is required. */
+  private static boolean EXACT_TYPE_MATCH;
+
+  /** If true, only return sequences that are appropriate to use as a method call receiver. */
+  private static boolean ONLY_RECEIVERS;
+
+  // TODO: The original paper uses a "secondary object pool (SequenceCollection in Randoop)"
+  // to store the results of the demand-driven input creation. This theorectically reduces
+  // the search space for the missing types. Consider implementing this feature and test whether
+  // it improves the performance.
+
+  /**
+   * Performs a demand-driven approach for constructing input objects of a specified type, when the
+   * sequence collection contains no objects of that type.
+   *
+   * <p>This method internally identifies a set of methods/constructors that return objects of the
+   * required type. For each of these methods: it generates a method sequence for the method by
+   * recursively searching for necessary inputs from the provided sequence collection; executes it;
+   * and if successful, stores the sequence in the sequence collection for future use.
+   *
+   * <p>Finally, it returns the newly-created sequences.
+   *
+   * <p>Invariant: This method is only called where the component manager lacks an object that is of
+   * a type compatible with the one required by the forward generator. See {@link
+   * randoop.generation.ForwardGenerator#selectInputs}.
+   *
+   * @param sequenceCollection the sequence collection from which to draw input sequences
+   * @param t the type of objects to create
+   * @param exactTypeMatch the flag to indicate whether an exact type match is required
+   * @param onlyReceivers if true, only return sequences that are appropriate to use as a method
+   *     call receiver
+   * @return method sequences that produce objects of the required type
+   */
+  public static SimpleList<Sequence> createInputForType(
+      SequenceCollection sequenceCollection,
+      Type t,
+      boolean exactTypeMatch,
+      boolean onlyReceivers) {
+
+    EXACT_TYPE_MATCH = exactTypeMatch;
+    ONLY_RECEIVERS = onlyReceivers;
+
+    // All constructors/methods found that return the demanded type.
+    Set<TypedOperation> producerMethods = getProducerMethods(t);
+
+    // For each producer method, create a sequence that produces an object of the demanded type
+    // if possible, or produce a sequence that leads to the eventual creation of the demanded type.
```

Comment:
I don't understand the distinction being made here.  Maybe the first one adds only one new method call and the second might add multiple.  Is the distinction important?